src/frontend/components/node-item/node-item.ts

```diff
@@ -1,18 +1,16 @@
-import {Component, Inject, NgZone, Input, SimpleChange} from '@angular/core';
+import {Component, NgZone, Input, ChangeDetectionStrategy} from '@angular/core';
 import {NgIf, NgFor} from '@angular/common';
-import * as Rx from 'rxjs';
-import {ComponentDataStore}
-  from '../../stores/component-data/component-data-store';
+import {ComponentDataStore} from
+  '../../stores/component-data/component-data-store';
 import {UserActions} from '../../actions/user-actions/user-actions';
-import {UserActionType}
-  from '../../actions/action-constants';
 
 // NOTE(cbond): This template must remain inline, there is a bug with recursive
 // controls in Angular 2 and templateUrl that prevents this from working
 // otherwise. Once they fix that bug then this can be placed back inside a
 // templateUrl file.
 @Component({
   selector: 'bt-node-item',
+  changeDetection: ChangeDetectionStrategy.OnPush,
```

Comment:
@ericjim let's discuss the use of change detection strategies outside the context of this PR. If I doesnt do anything at this point, I would remove it.